src/ostorlab/runtimes/local/models/models.py

```diff
@@ -412,18 +424,51 @@ def create(
         Returns:
             AgentArgument object.
         """
+
         agent_argument = AgentArgument(
             agent_id=agent_id,
             name=name,
             type=type,
             description=description,
-            value=value,
+            value=value
+            if isinstance(value, bytes) is True
+            else AgentArgument.to_bytes(type, value),
         )
         with Database() as session:
             session.add(agent_argument)
             session.commit()
             return agent_argument
 
+    @staticmethod
+    def to_bytes(
+        type: str, value: Optional[Union[int, float, str, bool, List, Dict]]
+    ) -> bytes:
+        """Convert the value to bytes."""
+        if type == "string":
+            return value.encode(encoding="utf-8")
+        elif type in ("boolean", "bool"):
+            return (1 if value is True else 0).to_bytes(1, byteorder="big")
+        elif type in ("number", "int", "float"):
```

Comment:
We only support `number`

Reply:
in the schema yes, but in some agents we use others. Example nmap:
![image](https://github.com/Ostorlab/oxo/assets/144013278/1f6e4ede-5d65-4b87-b6b3-a5708f482ab9)

Reply:
We do not force the type check when schema validation? how it was not spotted ..
The agent & the schema should be aligned..

Reply:
+1